daemon/algod/api/server/v2/handlers.go

```diff
@@ -310,6 +319,191 @@ func (v2 *Handlers) AccountInformation(ctx echo.Context, address string, params
+		PendingRewards:              pendingRewards.Raw,
+		AmountWithoutPendingRewards: amountWithoutPendingRewards.Raw,
+		Rewards:                     record.RewardedMicroAlgos.Raw,
+		Status:                      record.Status.String(),
+		RewardBase:                  &record.RewardsBase,
+		Participation:               apiParticipation,
+		TotalCreatedAssets:          record.TotalAssetParams,
+		TotalCreatedApps:            record.TotalAppParams,
+		TotalAssets:                 record.TotalAssets,
+		AuthAddr:                    addrOrNil(record.AuthAddr),
+		TotalAppsLocalState:         record.TotalAppLocalStates,
+		AppsTotalSchema: &generated.ApplicationStateSchema{
+			NumByteSlice: record.TotalAppSchema.NumByteSlice,
+			NumUint:      record.TotalAppSchema.NumUint,
+		},
+		AppsTotalExtraPages: numOrNil(uint64(record.TotalExtraAppPages)),
+		MinBalance:          record.MinBalance(&consensus).Raw,
+	}
+	response := generated.AccountResponse(account)
+	return ctx.JSON(http.StatusOK, response)
+}
+
+// AccountAssetInformation gets account information about a given asset.
+// (GET /v2/accounts/{address}/assets/{asset-id})
+func (v2 *Handlers) AccountAssetInformation(ctx echo.Context, address string, assetID uint64, params generated.AccountAssetInformationParams) error {
+	handle, contentType, err := getCodecHandle(params.Format)
+	if err != nil {
+		return badRequest(ctx, err, errFailedParsingFormatOption, v2.Log)
+	}
+
+	addr, err := basics.UnmarshalChecksumAddress(address)
+	if err != nil {
+		return badRequest(ctx, err, errFailedToParseAddress, v2.Log)
+	}
+
+	ledger := v2.Node.Ledger()
+
+	lastRound := ledger.Latest()
+	record, err := ledger.LookupResource(lastRound, addr, basics.CreatableIndex(assetID), basics.AssetCreatable)
+	if err != nil {
+		return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
+	}
+
+	if record.AssetParams == nil && record.AssetHolding == nil {
+		return notFound(ctx, errors.New(errAssetDoesNotExist), errAssetDoesNotExist, v2.Log)
+	}
+
+	// return msgpack response
+	if handle == protocol.CodecHandle {
+		data, err := encode(handle, record)
+		if err != nil {
+			return internalError(ctx, err, errFailedToEncodeResponse, v2.Log)
+		}
+		return ctx.Blob(http.StatusOK, contentType, data)
+	}
+
+	// prepare JSON response
+	response := generated.AccountAssetResponse{Round: uint64(lastRound)}
+
+	if record.AssetParams != nil {
+		asset := AssetParamsToAsset(addr.String(), basics.AssetIndex(assetID), record.AssetParams)
+		response.CreatedAsset = &asset.Params
+	}
+
+	if record.AssetHolding != nil {
+		// AccountInformation REST API endpoint and generated.AssetHolding provides creator address with AssetHolding
+		creatorAddr, ok, err := ledger.GetCreator(basics.CreatableIndex(assetID), basics.AssetCreatable)
+		var creator string
+		if err == nil && ok {
+			creator = creatorAddr.String()
+		} else {
+			// Asset may have been deleted, so we can no
+			// longer fetch the creator
+			creator = ""
```

Comment:
if the asset was deleted, then all the holding of that asset would have been deleted as well.
I'd suggest you'll respond with InternalServerError in this case.

Reply:
I copied this behavior from the current implementation of the `/v2/accounts/X` endpoint handler in https://github.com/algorand/go-algorand/blob/master/daemon/algod/api/server/v2/handlers.go#L288-L303 where it loops through all AssetHoldings to fill out a "creator" string for the response. I would love to get rid of that field though..

I believe it is possible to be a holder of a deleted asset however. You will still need to submit a transaction to close it out.

Reply:
given that this is a new entry point, we're not committed to support the same fields. I honestly can't tell if this is even something that would be of interest to a client.

I think that you can't be a holder to a deleted asset, but you could have opt-in for that asset.
( i.e. when deleting an asset, it checks that you hold all the instances of that asset )

Reply:
Right, you will still show up as opted in to hold a 0 balance of a deleted asset. There are lots of accounts in this situation and I think we would want to still show this as a holding for them.

Reply:
Otherwise they won't know what to do so they can opt out and lower their min balance. For example if you bought 1000 NFTs, then got rid of them, and then the creator deleted the asset, you would not even be able to see why your min balance was so high now.

Reply:
If the asset was destroyed, then they can't have non-zero asset amount.
But it's true that they would need to close-out to the creator address in order to opt-out and reduce their min balance.

Reply:
I removed the "creator" string from the AssetHolding response type, which I think resolves this